### DISCUSSION TITLE
"Database Administration Tool Extensions for Windows" roadmap for availability on MacOS

### DISCUSSION THREAD

**Discussion by spencer741** [microsoft/azuredatastudio#14124] [General]
When I heard about Azure Data Studio being cross-platform, I tried it out on my mac and was pretty impressed. Is there a roadmap for the "Database Administration Tool Extensions for Windows" extension being ported to work on MacOS?

Am I just not comprehending the intended scope with this sort of stuff?

I would like to get involved helping make this a possibility if it is not already being worked on .... is this possible with the current state of Azure Data Studio?

Thanks!

**Comment by Charles-Gagnon** [comment]
Unfortunately not at this time - that extension uses a minified version of the SSMS properties dialogs and as such is Windows only (the VS Shell that SSMS runs in is Windows only). And there are no current plans to create a similar native properties dialog in ADS.

But we'd happily accept contributions of a 3rd party extension with this behavior - let me know if you have any questions about how to get started and I'd be glad to provide some pointers!

**Comment by brandonweir** [comment]
@spencer741 - Did you happen to take up working on a contribution for this like @Charles-Gagnon suggested? I too would be really interested in seeing a "Database Admin Tool Extension for Mac OS" if anyone is working on this.

**Comment by spencer741** [comment]
@weirbt2k

I never did. If my memory serves me correct... For most of that functionality, they are invoking the SSMS screens that handle some database tooling. I don't believe those SSMS dialogs are open source. To rebuild 1:1 functionality, I guess you could start with the SSMS Dialogs to reference to build a base UI template, but functionality you'd have to build 100% yourself. 

I think that's why I got discouraged. Would be a pretty large project to take on by myself in the little free time I have.

Now that you have brought this to my attention again, I might see what I can cook up and start developing this extension for Mac... who knows, it might get some traction.
